include/core/io/CAN.hpp

```diff
@@ -128,15 +128,11 @@ class CAN {
      */
     static constexpr uint32_t DEFAULT_BAUD = 500000;
 
-private:
+protected:
```

Comment:
The port id is only used during initialization so it is a bit of a waste to hold onto it for the duration of the program especially since the pins that are a required part of the abstract class are used to derive that data. Additionally we are storing the pins in the abstract class and never using it, defeating the point of storing them in the first point. In general, I would say an abstract class members should be protected so that a derived class can properly extend the parent's functionality unless that member is critical to the parent's common behavior.

Reply:
This argument makes me think we should delete the `txPin` and `rxPin` from this class and just save port number in the concrete classes, but that may be more work than it's worth right now

Reply:
The `txPin` and `rxPin` are more useful than just a port number as they can be used to reconfigure gpio if necessary. Right now that isn't important, but with a different platform it may.